File: solutions/php/eliuds-eggs/1/EliudsEggs.php
<?php
declare(strict_types=1);
class EliudsEggs
{
    public function eggCount(int $displayValue): int
    {
        $numberOfEggs = 0;
        $highestPowerOfTwo = ceil(log($displayValue) / log(2));
        $remainder = $displayValue;
    
        while ($remainder > 0) {
            $binaryFactor = pow(2, $highestPowerOfTwo);
            if ($remainder >= $binaryFactor) {
                $numberOfEggs += 1;
                $remainder -= $binaryFactor;
            }
            $highestPowerOfTwo -= 1;
        }
    
        return $numberOfEggs;
    }
}

File: solutions/erlang/eliuds-eggs/1/src/eliuds_eggs.erl
-module(eliuds_eggs).

-export([egg_count/1]).


egg_count(0) ->
	0;
egg_count(N) ->
	N1 = N bsr 1,
	case N band 1 of
		1 ->
			1+egg_count(N1);
		0 ->
			egg_count(N1)
	end.